src/Adapter/ContextLoadHelper.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter;
+
+use Context;
+use Currency;
+use DummyController;
+use Employee;
+use Shop;
+
+/**
+ * Helps loading specific context, for example in CLI context
+ */
+class ContextLoadHelper
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param LegacyContext $context
+     */
+    public function __construct(LegacyContext $context)
```

Comment:
I'm wondering if we should rely on `LegacyContext` or inject `Context` (this can be done in the service definition anyway)
It's just that since `LegacyContext` is not really useful anyway besides getting access to the original `Context` we might as well let the class only dependable on what it needs It would be more flexible as it could be used in other use cases by injecting a pre-created context:

```php
$context = new Context();
// ... something happens
$loader = new ContextLoadHelper($context);
// ... load what you need
```

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/21125/commits/2b89cac34e6ebcf56b999f38faf4277a7b485d0f